public/app/features/admin/migrate-to-cloud/onprem/EmptyState/CallToAction/ConnectModal.tsx

```diff
@@ -0,0 +1,127 @@
+import { css } from '@emotion/css';
+import React, { useId, useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Modal, Button, Stack, TextLink, Field, Input, Text, useStyles2 } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
+
+import { ConnectStackDTO } from '../../../api';
+
+interface Props {
+  hideModal: () => void;
+  onConfirm: (connectStackData: ConnectStackDTO) => Promise<{ data: void } | { error: unknown }>;
+}
+
+export const ConnectModal = ({ hideModal, onConfirm }: Props) => {
+  const [isConnecting, setIsConnecting] = useState(false);
+  const cloudStackId = useId();
+  const tokenId = useId();
+  const styles = useStyles2(getStyles);
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    watch,
+  } = useForm<ConnectStackDTO>({
+    mode: 'onBlur',
```

Comment:
Imo if the only validation is that the field is required, doing it on blur is not needed.